### PR TITLE
Validate resending order confirmation email

### PR DESCRIPTION
**Pull request by TheMilek**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.13                   |
| Bug fix?        | no                                                       |
| New feature?    | yes                                                       |
| BC breaks?      | no                                                       |
| Deprecations?   | no  |
| License         | MIT                                                          |

Continuation of https://github.com/Sylius/Sylius/pull/15712 with validation added

### CODE REVIEW

File: src/Sylius/Bundle/OrderBundle/Resources/config/app/config.yml [hunk 1]
```diff
@@ -7,3 +7,12 @@ jms_serializer:
             sylius-order:
                 namespace_prefix: "Sylius\\Component\\Order"
                 path: "@SyliusOrderBundle/Resources/config/serializer"
+
+parameters:
```
Comment (Wojdylak) [inline src/Sylius/Bundle/OrderBundle/Resources/config/app/config.yml]: This is just a suggestion: the parameters before the jms_serializer.

**Review by Wojdylak — DISMISSED** [review]

**Review by Wojdylak — DISMISSED** [review]

**Review by Rafikooo — DISMISSED** [review]

File: src/Sylius/Bundle/CoreBundle/Validator/Constraints/ResendOrderConfirmationEmailWithValidOrderStateValidator.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        private RepositoryInterface $orderRepository,
+        private array $orderStatesToAllowResendingConfirmationEmail
+    ) {
+    }
+
+    public function validate(mixed $value, Constraint $constraint): void
+    {
+        if (!$value instanceof ResendOrderConfirmationEmail) {
+            throw new UnexpectedTypeException($value, ResendOrderConfirmationEmail::class);
+        }
```
Comment (jakubtobiasz) [inline src/Sylius/Bundle/CoreBundle/Validator/Constraints/ResendOrderConfirmationEmailWithValidOrderStateValidator.php:39]: It's not covered in the PHPSpec from what I see.

File: src/Sylius/Bundle/CoreBundle/Validator/Constraints/ResendOrderConfirmationEmailWithValidOrderStateValidator.php [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        if (!$constraint instanceof ResendOrderConfirmationEmailWithValidOrderState) {
+            throw new UnexpectedTypeException($constraint, ResendOrderConfirmationEmailWithValidOrderState::class);
+        }
+
+        /** @var OrderInterface|null $order */
+        $order = $this->orderRepository->findOneBy(['tokenValue' => $value->getOrderTokenValue()]);
+
+        if (null === $order) {
+            throw new NotFoundHttpException(sprintf('Order with %s token has not been found.', $value->getOrderTokenValue()));
+        }
```
Comment (jakubtobiasz) [inline src/Sylius/Bundle/CoreBundle/Validator/Constraints/ResendOrderConfirmationEmailWithValidOrderStateValidator.php]: As above + this layer shouldn't be aware about the "Http". Especially this exception is "general". Try to use (or create) meaningful exceptions telling what really went wrong. Here you might create (if it doesn't exist yet) the `OrderNotFoundException`.

**Review by jakubtobiasz — CHANGES_REQUESTED** [review]

**Review by jakubtobiasz — APPROVED** [review]

**Comment by jakubtobiasz** [comment]
Thank you, @TheMilek!

**Pull request merged**